addons/account/views/partner_view.xml

```diff
@@ -269,6 +272,9 @@
                    <filter string="Vendor Bills" name="supplier" domain="[('supplier_rank','>', 0)]"/>
                    <separator/>
                 </xpath>
+                <xpath expr="//field[@name='name']" position="replace">
+                    <field name="name" filter_domain="['|', '|', '|', '|', ('display_name', 'ilike', self), ('ref', '=', self), ('email', 'ilike', self), ('vat', 'ilike', self), ('company_registry_number', 'ilike', self)]"/>
+                </xpath>
```

Comment:
`position="replace"` on a field like this :fearful: 
Maybe have a look at `_rec_names_search`